src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -52,12 +52,18 @@ namespace CDASimAdapter{
 
         //Read local sensor file and populate the sensors JSON
         //Sample sensors.json: https://raw.githubusercontent.com/usdot-fhwa-OPS/V2X-Hub/develop/src/v2i-hub/CDASimAdapter/test/sensors.json
-        auto sensors_json_v = read_json_file(_sensor_json_file_path);
-        if(sensors_json_v.empty())
-        {
-            PLOG(logWARNING) << "Sensors JSON is empty!" << std::endl;
-        }     
-        message["sensors"] = sensors_json_v;
+        // Sensor configuration is an optional part of registration message.
+        if ( !_sensor_json_file_path.empty() ) {
+            auto sensors_json_v = read_json_file(_sensor_json_file_path);
+            if(sensors_json_v.empty())
+            {
+                PLOG(logWARNING) << "Sensors JSON is empty!" << std::endl;
+            }     
+            message["sensors"] = sensors_json_v;
```

Comment:
Do we need to set this sensors in message? I noticed that if the path to sensor json file is empty, this sensors is not set in message.

Reply:
Yep it is an optional part of the registration message. When included it configures simulated sensor for V2X-Hub/CARMA-Streets. Since V2X-Hub/CARMA-Streets basic functionality does not require sensors it is also not required.

Reply:
I get the optional part. I am trying to make a note that whether we need to set ` message["sensors"] = sensors_json_v;` when it is empty because I see it is not set when the path is empty.

Reply:
We do not need the that part when the sensor configuration file is empty or missing